src/GraphQLParser/AST/ASTListNode.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GraphQLParser.AST
+{
+    /// <summary>
+    /// Represents a AST node that holds a list of other (nested) AST nodes.
+    /// </summary>
+    public abstract class ASTListNode<TNode> : ASTNode, IEnumerable<TNode>
```

Comment:
```suggestion
    public abstract class ASTListNode<TNode> : ASTNode, IReadOnlyList<TNode>
```
No other changes needed here (but probably api approvals would change)

Reply:
done